---
title: "ICE-A: Figure 2"
output: 
  html_notebook
    #toc: true
---
# 2. B/T elements

## 2.0 Preperations
Run before:
conda activate icea_analysis_env
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/anaconda3/envs/icea_analysis_env/lib

Setting working directory
```{r setup}
knitr::opts_knit$set(root.dir = normalizePath("../../"))
```


Loading packages
```{r}
#Loading libraries
packages <- c("tidyr", "dplyr", "stringr", "DESeq2", "tximport", "AnnotationDbi", "org.Mm.eg.db", "circlize", "viridis", "ComplexHeatmap", "clusterProfiler", "DiffBind", "ggplot2", "ggpubr")
lapply(packages, library, character.only = TRUE)

```


## 2.1 Identification of B/T gene list
### 2.1.1 Differntial genes FrBC vs DN2b
```{r}
#Import Immgen RNAseq data (preprocessed with: rnaseq_star_rsem_PE_sample_processing.skript)
files.rna.immgen <- list.files(path = "data/rna/immgen/counts", pattern = "genes.results$", full.names = T)
names(files.rna.immgen) <- sub(".Aligned.toTranscriptome.rsem.out.genes.results", "",list.files(path = "data/rna/immgen/counts", pattern = "genes.results$"))
names(files.rna.immgen) <- c(names(files.rna.immgen[1:15]), "LTHSC_34neg_1", "LTHSC_34neg_2", "LTHSC_34pos_1", "LTHSC_34pos_2", names(files.rna.immgen[20:23]))
files.rna.immgen <- files.rna.immgen[c(16:19, 22:23, 20:21, 1:2, 10:15, 3:9)]
txi.rna.rna.immgen <- tximport(files.rna.immgen, type="rsem", txIn = F, txOut = F, importer = read.delim)

#Convert to entrezID & filter
counts.rna.immgen <- data.frame(ensembl=str_replace(rownames(txi.rna.rna.immgen$counts), pattern="\\.[0-9]+_.*$", replacement = ""), txi.rna.rna.immgen$counts)
counts.rna.immgen$entrezID <- mapIds(org.Mm.eg.db, keys = counts.rna.immgen$ensembl, column = "ENTREZID", keytype = "ENSEMBL")
counts.rna.immgen <- aggregate(. ~entrezID, data=counts.rna.immgen[,-1], sum)
rownames(counts.rna.immgen) <- counts.rna.immgen$entrezID
counts.rna.immgen <- counts.rna.immgen[,-1]
counts.rna.immgen <- round(counts.rna.immgen, 0)
counts.rna.immgen <- counts.rna.immgen[rowSums(counts.rna.immgen >= 10) >= 2,]

#Create coldata
coldata.rna.immgen <- data.frame(sampleID=colnames(counts.rna.immgen), celltype=c(rep(c("LTHSC_34neg","LTHSC_34pos","STHSC","MPP4","CLP","FrA","FrBC", "FrE", "DN1"),each=2),"DN2a", rep(c("DN2b", "DN3"), each=2)), replicate=c(rep(c(1,2), 9), 1, 1,2,1,2))
coldata.rna.immgen$celltype <- factor(coldata.rna.immgen$celltype, levels = c("LTHSC_34neg","LTHSC_34pos","STHSC","MPP4","CLP","FrA","FrBC", "FrE", "DN1","DN2a", "DN2b", "DN3"))

#Save data
save(counts.rna.immgen, file="output/rdata/figure2/counts.rna.immgen.Rdata")
load(file="output/rdata/figure2/counts.rna.immgen.Rdata")
save(coldata.rna.immgen, file="output/rdata/figure2/coldata.rna.immgen.Rdata")
load(file="output/rdata/figure2/coldata.rna.immgen.Rdata")

#Identify differntially expressed genes between FrBC and DN2b
source("script/general/r_functions/DESeq2_fun.R")
DESeq2_fun(counts = counts.rna.immgen, coldata = coldata.rna.immgen, selection = c(13,14,20,21), factor="celltype", groups = c("DN2b", "FrBC"), filt=T, filt_n = 2, filt_value = 10, mapID = T, orgdb = org.Mm.eg.db)
DE.FrBCvsDN2b.result.diff <- DE.FrBCvsDN2b.result.df[abs(DE.FrBCvsDN2b.result.df$log2FoldChange)> 2 & !is.na(DE.FrBCvsDN2b.result.df$padj) & DE.FrBCvsDN2b.result.df$padj<0.01,]
```

### 2.1.2 Clustering of differntial genes within lymmphoid popualtions
```{r}
#Identify clusters of differential expressed genes
dds.rna.immgen <- DESeqDataSetFromMatrix(countData = counts.rna.immgen, colData = coldata.rna.immgen, design = ~ celltype)
dds.rna.immgen <- DESeq(dds.rna.immgen)
vst.rna.immgen <- vst(dds.rna.immgen)
vst.rna.immgen.df <- data.frame(assay(vst.rna.immgen))

vst.rna.immgen.mean <- data.frame(LTHSC_34neg=rowMeans(vst.rna.immgen.df[,1:2]), LTHSC_34pos=rowMeans(vst.rna.immgen.df[,3:4]), STHSC=rowMeans(vst.rna.immgen.df[,5:6]), MPP4=rowMeans(vst.rna.immgen.df[,7:8]),CLP=rowMeans(vst.rna.immgen.df[,9:10]),FrA=rowMeans(vst.rna.immgen.df[,11:12]), FrBC=rowMeans(vst.rna.immgen.df[,13:14]), FrE=rowMeans(vst.rna.immgen.df[,15:16]), DN1=rowMeans(vst.rna.immgen.df[,17:18]),DN2a=vst.rna.immgen.df[,19], DN2b=rowMeans(vst.rna.immgen.df[,20:21]), DN3=rowMeans(vst.rna.immgen.df[,22:23]))
vst.rna.immgen.z <- as.data.frame(vst.rna.immgen.mean-rowMeans(vst.rna.immgen.mean))/rowSds(as.matrix(vst.rna.immgen.mean))
vst.rna.immgen.z <- na.omit(vst.rna.immgen.z)
save(vst.rna.immgen.z, file="output/rdata/figure2/vst.rna.immgen.z.Rdata")
vst.rna.immgen.z.BT <- vst.rna.immgen.z[rownames(vst.rna.immgen.z) %in% DE.FrBCvsDN2b.result.diff$row, ]

groups = c(rep("Common progenitors", 4), rep( "B cell progenitors",4), rep( "T cell progenitors",4))
groups <- factor(groups, levels = c( "Common progenitors", "T cell progenitors", "B cell progenitors"))
set.seed(12)
col <- colorRamp2(c(-3.5, -1.75, 0, 1.75, 3.5),colors = c("#1f426a", "#8f9bb2", "white", "#b58ea7", "#6a2456"))
BT.heatmap <- Heatmap(vst.rna.immgen.z.BT, col=col, name="BT", row_km = 4, show_row_names=F, cluster_columns = F, column_split=groups, show_row_dend = F, top_annotation = HeatmapAnnotation(Group = anno_block(gp = gpar(fill = c("#818284", "#1f426a", "#6a2456"))), height = unit(0.2, "cm")), show_parent_dend_line = F, column_title = NULL, column_names_gp = gpar(fontsize=6),heatmap_legend_param = list(title="z-score",title_gp=gpar(fontsize=8, fontface="bold"), labels_gp=gpar( fontsize=6)), row_title_gp = gpar( fontsize=8, fontface="bold"), raster_device="CairoPNG")

set.seed(4)
BT.heatmap <-draw(BT.heatmap, main_heatmap = "BT", heatmap_width = unit(12, "cm"), heatmap_height = unit(8, "cm")) 

cairo_pdf("output/plots/figure2/B_T_genes_heatmap.pdf")
BT.heatmap
dev.off()

#Extracting clusters
BT.cluster.list <- row_order(BT.heatmap) 
BT.clusters <- lapply(names(BT.cluster.list), function(i){
   out <- data.frame(entrezID = rownames(vst.rna.immgen.z.BT[BT.cluster.list[[i]],]),
                     Cluster = paste0("cluster", i),
                     stringsAsFactors = FALSE)
   rownames(out) <- out$Clusters
   return(out)
}) %>%
do.call(rbind, .)
BT.clusters$symbol <-mapIds(org.Mm.eg.db, keys=BT.clusters$entrezID, column="SYMBOL", keytype="ENTREZID")

#Gene lists
T.genes <- BT.clusters[BT.clusters$Cluster=="cluster4",]
write.table(T.genes[,3], "output/data/figure2/genelists/Tcell_genes.txt", quote = F, sep = "\t", row.names = F, col.names = F)
B.genes <- BT.clusters[BT.clusters$Cluster=="cluster2",]
write.table(B.genes[,3], "output/data/figure2/genelists/Bcell_genes.txt", quote = F, sep = "\t", row.names = F, col.names = F)

```



## 2.2 Identification of B/T putative enhancers map

### 2.2.1 Define consensus peak set for element identification
Merged IDR set of 230238/P2C2 used
```{bash engine.opts='-l'}
cd output/data/figure2/peaks/ 
bash ../../../../script/general/peak_union.sh /mnt/data/common/tobias/tg/ICE_A/ICEA_analysis/data/atac/peaks/230238_atac.idr.optimal_peak.narrowPeak /mnt/data/common/tobias/tg/ICE_A/ICEA_analysis/data/atac/peaks/p2c2_atac_idr.optimal_peak.narrowPeak 230238 P2C2 atac_230238_p2c2_idr_consensus_info.bed
awk -v FS='\t' -v OFS='\t' '{print $1, $2, $3, $1":"$2"-"$3}' atac_230238_p2c2_idr_consensus_info.bed > atac_230238_p2c2_idr_consensus.bed
cut -f1-3 atac_230238_p2c2_idr_consensus.bed > atac_230238_p2c2_idr_consensus_cut.bed
```


### 2.2.2 Annotation
ICE-A annotation using 230238/P2C2 PLAC-seq data
```{bash engine.opts='-l'}
script_dir="script/figure2"
bash $script_dir/icea_230238_p2c2_atac_consensus_annotation.sh
```

Organize annotation output
```{r}
dist <- 7500

#LIANA - 230238
BT_elements_anno_ICEA230238 <- read.table("output/data/figure2/annotation/atac_230238_p2c2_consensus_peaks_230238_H3K4me3_PLAC_annotation/Peak_annotation/atac_230238_p2c2_consensus/atac_230238_p2c2_consensus_PLACseq_annotated.txt", header=TRUE, sep="\t")
BT_elements_anno_ICEA230238$peakID <- paste0(BT_elements_anno_ICEA230238$Chr, ":", BT_elements_anno_ICEA230238$Start, "-", BT_elements_anno_ICEA230238$End)
BT_elements_anno_ICEA230238 <- BT_elements_anno_ICEA230238[!is.na(BT_elements_anno_ICEA230238$Distance_to_TSS),]
BT_elements_anno_ICEA230238$Type <- ifelse(abs(BT_elements_anno_ICEA230238$Distance_to_TSS) <= dist, "Proximal", "Distal")
BT_elements_anno_ICEA230238 <- BT_elements_anno_ICEA230238[,c(17,9,5,18)]
BT_elements_anno_ICEA230238 <- BT_elements_anno_ICEA230238[!duplicated(BT_elements_anno_ICEA230238),]

#ICEA - P2C2
BT_elements_anno_ICEAP2C2 <- read.table("output/data/figure2/annotation/atac_230238_p2c2_consensus_peaks_P2C2_H3K4me3_PLAC_annotation/Peak_annotation/atac_230238_p2c2_consensus/atac_230238_p2c2_consensus_PLACseq_annotated.txt", header=TRUE, sep="\t")
BT_elements_anno_ICEAP2C2$peakID <- paste0(BT_elements_anno_ICEAP2C2$Chr, ":", BT_elements_anno_ICEAP2C2$Start, "-", BT_elements_anno_ICEAP2C2$End)
BT_elements_anno_ICEAP2C2 <- BT_elements_anno_ICEAP2C2[!is.na(BT_elements_anno_ICEAP2C2$Distance_to_TSS),]
BT_elements_anno_ICEAP2C2$Type <- ifelse(abs(BT_elements_anno_ICEAP2C2$Distance_to_TSS) <= dist, "Proximal", "Distal")
BT_elements_anno_ICEAP2C2 <- BT_elements_anno_ICEAP2C2[,c(17,9,5,18)]
BT_elements_anno_ICEAP2C2 <- BT_elements_anno_ICEAP2C2[!duplicated(BT_elements_anno_ICEAP2C2),]
```


### 3.2.3 Identification of putative regualatory element of B/T genes
```{r}
BT.elements.ICEA <- rbind(data.frame(BT_elements_anno_ICEA230238[BT_elements_anno_ICEA230238$Gene %in% B.genes$symbol,], group="B"), data.frame(BT_elements_anno_ICEAP2C2[BT_elements_anno_ICEAP2C2$Gene %in% T.genes$symbol,], group="T"))
save(BT.elements.ICEA, file="output/rdata/figure2/BT_elements_liana.Rdata")

#Filter if not open
BT_elements_info <- read.table("output/data/figure2/peaks/atac_230238_p2c2_idr_consensus_info.bed", header=F, sep="\t")
colnames(BT_elements_info) <- c("chr", "start", "end", "celltype")
BT_elements_info$peakID <- paste0(BT_elements_info$chr, ":", BT_elements_info$start, "-", BT_elements_info$end)

BT.elements.ICEA <- BT.elements.ICEA[!(BT.elements.ICEA$group=="B" & BT.elements.ICEA$peakID %in% BT_elements_info[BT_elements_info$celltype=="P2C2",]$peakID) & !(BT.elements.ICEA$group=="T" & BT.elements.ICEA$peakID %in% BT_elements_info[BT_elements_info$celltype=="230238",]$peakID),]
```



## 2.3 Chromatin status in B/T elements

### 2.3.1 Preprocessing 
```{r}
#Loading sample info
samples.atac.230238.p2c2 <- read.table("data/atac/info/samples_atac_230238_p2c2.txt", header=TRUE, sep="\t", quote = "")

#Extracting normalized coutns
dbo.atac.230238.p2c2 <- dba(sampleSheet=samples.atac.230238.p2c2)
counts.atac.230238.p2c2 <- dba.count(dbo.atac.230238.p2c2, peaks="output/data/figure2/peaks/atac_230238_p2c2_idr_consensus_cut.bed", summits = F)
save(counts.atac.230238.p2c2, file="output/rdata/figure2/counts.atac.230238.p2c2_atac_230238_p2c2_consensus_peaks.Rdata")

#Saving raw counts from consensus peak set to data frame
atac.230238.p2c2.raw.count <- lapply(counts.atac.230238.p2c2[["peaks"]],function(x) x[,6])
atac.230238.p2c2.raw.count.df<- as.data.frame(atac.230238.p2c2.raw.count)
rownames(atac.230238.p2c2.raw.count.df) <- paste0(counts.atac.230238.p2c2[["peaks"]][[1]][,1], ":", counts.atac.230238.p2c2[["peaks"]][[1]][,2], "-", counts.atac.230238.p2c2[["peaks"]][[1]][,3])
colnames(atac.230238.p2c2.raw.count.df) <- counts.atac.230238.p2c2$SampleID

#Extracting noramlized counts
counts.atac.230238.p2c2.norm <- dba.normalize(counts.atac.230238.p2c2)
counts.atac.230238.p2c2.norm <- dba.peakset(counts.atac.230238.p2c2, bRetrieve=TRUE)
counts.atac.230238.p2c2.norm.df <- as.data.frame(counts.atac.230238.p2c2.norm)
counts.atac.230238.p2c2.norm.df$peakID <- paste0(counts.atac.230238.p2c2.norm.df$seqnames, ":", counts.atac.230238.p2c2.norm.df$start, "-", counts.atac.230238.p2c2.norm.df$end)
counts.atac.230238.p2c2.norm.df.mean <- data.frame(peakID=counts.atac.230238.p2c2.norm.df$peakID, ATAC_230238=rowMeans(counts.atac.230238.p2c2.norm.df[,6:7]), ATAC_p2c2=rowMeans(counts.atac.230238.p2c2.norm.df[,8:9]))
write.table(counts.atac.230238.p2c2.norm.df.mean, "output/data/figure2/atac_230238_p2c2_consensus_peaks_count.txt", quote = F, sep = "\t", row.names = F, col.names = T)

#Convert to long format
counts.atac.230238.p2c2.norm.df.mean.long <- rbind(data.frame(peakID=counts.atac.230238.p2c2.norm.df.mean$peakID, ATAC=counts.atac.230238.p2c2.norm.df.mean$ATAC_230238, celltype="230238"),data.frame(peakID=counts.atac.230238.p2c2.norm.df.mean$peakID, ATAC=counts.atac.230238.p2c2.norm.df.mean$ATAC_p2c2, celltype="P2C2"))
```

### 3.3.2 Extract ATAC levels in B/T elements
```{r}
BT.elements.atac <- merge(BT.elements.ICEA, counts.atac.230238.p2c2.norm.df.mean.long)
BT.elements.atac$id <- paste0(BT.elements.atac$peakID, "_", BT.elements.atac$Gene)

#Filter
BT.elements.atac.filt <- BT.elements.atac[(BT.elements.atac$id %in% BT.elements.atac[BT.elements.atac$group=="B" & BT.elements.atac$celltype=="230238",]$id) | (BT.elements.atac$id %in% BT.elements.atac[BT.elements.atac$group=="T" & BT.elements.atac$celltype=="P2C2",]$id),]
```

### 3.3.3 Violin plot of atac levels
```{r}
my_comparisons <- list( c("230238", "P2C2"))

BT_element_atac <- ggplot(BT.elements.atac.filt[BT.elements.atac.filt$Type=="Distal",], aes(celltype,log2(ATAC+1), fill=celltype))+
  geom_violin()+ geom_boxplot(width=0.1, outlier.alpha = 0)+
  theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))+scale_fill_manual(values=c(rep(c("white","grey"), 6)))+
    facet_grid(group  ~ .)+
  ylim(0,13.5)+
  stat_compare_means(comparisons = my_comparisons, label.y = c(12), label="p.signif")

ggsave(path="output/plots/figure2", filename="BT_element_ATAC.pdf", plot = BT_element_atac, device = "pdf", dpi=300, width = 6, height = 4)
```


## 2.4 Investigation of B/T elements

### 2.4.1 Extract B/T enhancer map
Distal elements >2.5kb from TSS, annotated to B genes (230238 PLACseq) or T genes (P2C2 PLACseq)
```{r}
B.enh <- BT.elements.atac.filt[BT.elements.atac.filt$Type=="Distal" & BT.elements.atac.filt$group=="B" & BT.elements.atac.filt$celltype=="230238",]
B.enh.bed <- B.enh %>% separate(peakID, c("chr", "start", "end"), ":|-")
write.table(B.enh.bed[,1:3], "output/data/figure2/peaks/B_enhancers.bed", quote = F, sep = "\t", row.names = F, col.names = F)

T.enh <- BT.elements.atac.filt[BT.elements.atac.filt$Type=="Distal" & BT.elements.atac.filt$group=="T" &  BT.elements.atac.filt$celltype=="P2C2",]
T.enh.bed <- T.enh %>% separate(peakID, c("chr", "start", "end"), ":|-")
write.table(T.enh.bed[,1:3], "output/data/figure2/peaks/T_enhancers.bed", quote = F, sep = "\t", row.names = F, col.names = F)
```


### 2.4.2 Motif enrichemnt in B/T elements

HOMER
```{bash engine.opts='-l'}
cd output/data/figure2/motifs

source /opt/anaconda3/bin/activate ICE_A_env

findMotifsGenome.pl ../peaks/B_enhancers.bed mm10 B_enahncers_motifs_out -size 200 -mask

findMotifsGenome.pl ../peaks/T_enhancers.bed mm10 T_enahncers_motifs_out -size 200 -mask

```


## 2.5 Investigation of TF biding

### 2.5.1 LIANA multiple mode
```{bash engine.opts='-l'}
script_dir="script/figure2"
bash $script_dir/icea_multiple_tf_occupancy.sh
```


Percent binding
```{r}
#Calculate % biding for each factor of total regions annotated to selected gene list

#B
upset_b <- read.table("output/data/figure2/tf_occupancy/tf_occupancy_Bgenes_230238_H3K4me3_PLAC_annotation/tmp/process10/UpSet_PLACseq_interactions_Distal_genes.txt", header=T, sep="\t")
length(unique(upset_b[upset_b$Source=="EBF1",]$Target))/length(unique(upset_b$Target))
length(unique(upset_b[upset_b$Source=="PAX5",]$Target))/length(unique(upset_b$Target))
length(unique(upset_b[upset_b$Source=="TCF7",]$Target))/length(unique(upset_b$Target))
length(unique(upset_b[upset_b$Source=="GATA3",]$Target))/length(unique(upset_b$Target))

#T
upset_t <- read.table("output/data/figure2/tf_occupancy/tf_occupancy_Tgenes_P2C2_H3K4me3_PLAC_annotation/tmp/process10/UpSet_PLACseq_interactions_Distal_genes.txt", header=T, sep="\t")
length(unique(upset_t[upset_t$Source=="EBF1",]$Target))/length(unique(upset_t$Target))
length(unique(upset_t[upset_t$Source=="PAX5",]$Target))/length(unique(upset_t$Target))
length(unique(upset_t[upset_t$Source=="TCF7",]$Target))/length(unique(upset_t$Target))
length(unique(upset_t[upset_t$Source=="GATA3",]$Target))/length(unique(upset_t$Target))
```
## 2.6 Save files for supplementary
Filter output from TF occupancy analysis to only include regions in consensus peak list and open in celltype + target gene in B/T gene list respectively.
```{r}
# B elements (supp table 1C)
tf_anno_B <- read.table("/mnt/data/common/tobias/tg/ICE_A/ICEA_analysis/output/data/figure2/tf_occupancy/tf_occupancy_Bgenes_230238_H3K4me3_PLAC_annotation/Peak_annotation/ALL/ALL_PLACseq_annotated.txt", header=T, sep="\t")
tf_anno_B$peakID <- paste0(tf_anno_B$Chr, ":", tf_anno_B$Start, "-", tf_anno_B$End)
tf_anno_B <- tf_anno_B[tf_anno_B$peakID %in% BT_elements_info[BT_elements_info$celltype!="P2C2",]$peakID,]
tf_anno_B_filt <- tf_anno_B[tf_anno_B$Gene %in% B.genes$symbol,]
write.table(tf_anno_B_filt[,c(21,1:20)], "output/data/figure2/annotation/Supp_Table_1C_Bcell_elements_with_B_target_genes.txt", quote = F, sep = "\t", row.names = F, col.names = T)

# T elements (supp table 1D)
tf_anno_T <- read.table("/mnt/data/common/tobias/tg/ICE_A/ICEA_analysis/output/data/figure2/tf_occupancy/tf_occupancy_Tgenes_P2C2_H3K4me3_PLAC_annotation/Peak_annotation/ALL/ALL_PLACseq_annotated.txt", header=T, sep="\t")
tf_anno_T$peakID <- paste0(tf_anno_T$Chr, ":", tf_anno_T$Start, "-", tf_anno_T$End)
tf_anno_T <- tf_anno_T[tf_anno_T$peakID %in% BT_elements_info[BT_elements_info$celltype!="230238",]$peakID,]
tf_anno_T_filt <- tf_anno_T[tf_anno_T$Gene %in% T.genes$symbol,]
write.table(tf_anno_T_filt[,c(21,1:20)], "output/data/figure2/annotation/Supp_Table_1D_Tcell_elements_with_T_target_genes.txt", quote = F, sep = "\t", row.names = F, col.names = T)

```

